tests/std/tests/GH_005244_regex_escape_sequences/test.cpp

```diff
@@ -0,0 +1,882 @@
+    g_regexTester.should_match("\111", R"(\111)", awk);
+    g_regexTester.should_not_match("1", R"(\111)", awk);
+    g_regexTester.should_not_match("11", R"(\111)", awk);
+    g_regexTester.should_not_match("111", R"(\111)", awk);
+    g_regexTester.should_not_match("\\111", R"(\111)", awk);
+    // clang-format off
+    g_regexTester.should_match("\111""1", R"(\1111)", awk);
+    // clang-format on
+    g_regexTester.should_not_match("\111", R"(\1111)", awk);
+    g_regexTester.should_not_match("1", R"(\1111)", awk);
+    g_regexTester.should_not_match("11", R"(\1111)", awk);
+    g_regexTester.should_not_match("111", R"(\1111)", awk);
+    g_regexTester.should_not_match("1111", R"(\1111)", awk);
+    g_regexTester.should_not_match("\\1111", R"(\1111)", awk);
+    g_regexTester.should_match("A", R"(\101)", awk);
+    g_regexTester.should_match("Aa", R"(\101a)", awk);
+    g_regexTester.should_match("\33", R"(\033)", awk);
+    g_regexTester.should_match("\33a", R"(\033a)", awk);
+    g_regexTester.should_match("\33", R"(\33)", awk);
+    g_regexTester.should_match("\33a", R"(\33a)", awk);
+    g_regexTester.should_match("\1", R"(\001)", awk);
+    g_regexTester.should_match("\1a", R"(\001a)", awk);
+    g_regexTester.should_match("\1", R"(\01)", awk);
+    g_regexTester.should_match("\1a", R"(\01a)", awk);
+    g_regexTester.should_match("\1", R"(\1)", awk);
+    g_regexTester.should_match("\1a", R"(\1a)", awk);
+    g_regexTester.should_throw(R"(\8)", error_escape, awk);
+    // clang-format off
+    g_regexTester.should_match("\1""8", R"(\18)", awk);
+    g_regexTester.should_match("\12""9", R"(\129)", awk);
+    // clang-format on
+
+    // octal sequences evaluating to 0 are considered undefined by the standard
+    g_regexTester.should_throw(R"(\0)", error_escape, awk);
+    g_regexTester.should_throw(R"(\00)", error_escape, awk);
+    g_regexTester.should_throw(R"(\000)", error_escape, awk);
+}
+
+void test_gh_5244_classescape_posix_not_awk(syntax_option_type option) {
+    // Sections "BRE Bracket Expressions" and "ERE Bracket Expressions":
+    // Backslash shall lose its special meaning, so \c should always match \ + c
+
+    // common special characters outside character classes
+    check_classescape_noescape(".", option);
+    check_classescape_noescape("[", option);
+    check_classescape_noescape("*", option);
+    check_classescape_noescape("^", option);
+    check_classescape_noescape("$", option);
+    check_classescape_noescape("B", option);
+    check_classescape_noescape("c", option);
+
+    // special characters outside character classes in extended regexes
+    check_classescape_noescape("+", option);
+    check_classescape_noescape("?", option);
+    check_classescape_noescape("|", option);
+
+    // TRANSITION, GH-5379
+    if (option & (extended | egrep | awk)) {
+        check_classescape_noescape("(", option);
+        check_classescape_noescape(")", option);
+        check_classescape_noescape("{", option);
+    }

```

Comment:
We're in `test_gh_5244_classescape_posix_not_awk`, so we don't need to test for awk. And in fact, this is forming `string pattern = "[\\" + c + "]";`, which awk will throw exceptions for (as tested in the awk-specific function).